core/config/default.ts

```diff
@@ -61,21 +61,18 @@ export const DEFAULT_MODEL_GRANITE_LARGE: ModelConfig = {
   ...BASE_GRANITE_CONFIG,
 };
 
-export const DEFAULT_GRANITE_COMPLETION_MODEL_LARGE: ModelConfig = {
-  ...DEFAULT_MODEL_GRANITE_LARGE,
-  name: DEFAULT_MODEL_GRANITE_LARGE.name + "::autocomplete",
-  defaultCompletionOptions: {
-    ...DEFAULT_MODEL_GRANITE_LARGE.defaultCompletionOptions,
-    maxTokens: 100,
-  },
-  roles: ["autocomplete"],
-};
-
-export const DEFAULT_GRANITE_COMPLETION_MODEL_SMALL: ModelConfig = {
-  ...DEFAULT_MODEL_GRANITE_SMALL,
-  name: DEFAULT_MODEL_GRANITE_SMALL.name + "::autocomplete",
+// We optimize for speed over quality by using the 2b-base
+// model which is *almost* as good as the 8b-instruct model,
+// though not as good as the 8b-base model.
+export const DEFAULT_GRANITE_COMPLETION_MODEL: ModelConfig = {
+  ...BASE_GRANITE_CONFIG,
+  name: "granite3.3:2b-base",
+  provider: "ollama",
+  model: "ibm/granite3.3:2b-base",
   defaultCompletionOptions: {
-    ...DEFAULT_MODEL_GRANITE_SMALL.defaultCompletionOptions,
+    ...BASE_GRANITE_CONFIG.defaultCompletionOptions,
+    // This needs to be bigger than maxPromptTokens (1024) + maxTokens (100)
+    contextLength: 1536,
```

Comment:
why not DEFAULT_CONTEXT_LENGTH ? is the concern about memory overhead?

Reply:
Yes. KV cache size for granite3.3:2b is 1.25GB for num_ctx=16k, then typically multiply that by the default OLLAMA_NUM_PARALLEL and we're taking memory usage from 2.6GB to 7.6GB. And there's no need - we will never have more than 1124 tokens for input+output (1536 is just being a bit conservative).